pallets/capacity/src/tests.rs

```diff
@@ -770,3 +772,196 @@ fn calculate_capacity_reduction_determines_the_correct_capacity_reduction_amount
 
 	assert_eq!(capacity_reduction, 180);
 }
+
+#[test]
+fn impl_balance_is_successful() {
+	new_test_ext().execute_with(|| {
+		let target_msa_id = 1;
+		let remainging_amount = 10u32;
+		let total_available_amount = 10u32;
+
+		let _ = create_capacity_account_and_fund(
+			target_msa_id,
+			remainging_amount.into(),
+			total_available_amount.into(),
+		);
+
+		assert_eq!(Capacity::balance(target_msa_id), BalanceOf::<Test>::from(10u32));
+	});
+}
+
+#[test]
+fn impl_balance_returns_zero_when_target_capacity_is_not_found() {
+	new_test_ext().execute_with(|| {
+		let msa_id = 1;
+		assert_eq!(Capacity::balance(msa_id), BalanceOf::<Test>::zero());
+	});
+}
+
+#[test]
+fn impl_deduct_is_successful() {
+	new_test_ext().execute_with(|| {
+		let target_msa_id = 1;
+		let remaining_amount = BalanceOf::<Test>::from(10u32);
+		let total_available_amount = BalanceOf::<Test>::from(10u32);
+		let _ = create_capacity_account_and_fund(
+			target_msa_id,
+			remaining_amount,
+			total_available_amount,
+		);
+
+		assert_ok!(Capacity::deduct(target_msa_id, 5u32.into()));
+
+		let mut capacity_details = CapacityDetails::<
+			BalanceOf<Test>,
+			<Test as frame_system::Config>::BlockNumber,
+		>::default();
+
+		capacity_details.remaining = 5u32.into();
+		capacity_details.total_tokens_staked = 10u32.into();
+		capacity_details.total_available = 10u32.into();
+		capacity_details.last_replenished_epoch = 1u32.into();
+
+		assert_eq!(Capacity::get_capacity_for(target_msa_id).unwrap(), capacity_details);
+	});
+}
+
+#[test]
+fn impl_deduct_errors_target_capacity_not_found() {
+	new_test_ext().execute_with(|| {
+		let target_msa_id = 1;
+		let amount = BalanceOf::<Test>::from(10u32);
+		assert_noop!(
+			Capacity::deduct(target_msa_id, amount),
+			Error::<Test>::TargetCapacityNotFound
+		);
+	});
+}
+
+#[test]
+fn impl_deduct_errors_insufficien_balance() {
+	new_test_ext().execute_with(|| {
+		let target_msa_id = 1;
+		let remaining_amount = BalanceOf::<Test>::from(10u32);
+		let total_available_amount = BalanceOf::<Test>::from(10u32);
+		let _ = create_capacity_account_and_fund(
+			target_msa_id,
+			remaining_amount,
+			total_available_amount,
+		);
+
+		assert_noop!(
+			Capacity::deduct(target_msa_id, 20u32.into()),
+			Error::<Test>::InsufficientBalance
+		);
+
+		let mut capacity_details = CapacityDetails::<
```

Comment:
This is checking that trying to deduct more than is there doesn't alter the state?

Reply:
Yes.